consensus/src/block_storage/sync_manager.rs

```diff
@@ -351,25 +415,33 @@ impl BlockStore {
         );
 
         if !order_vote_enabled {
+            // TODO: this is probably still necessary, but need to think harder, it's pretty subtle
```

Comment:
I guess we're just going to remove this branch

Reply:
I think we're keeping it for now? cc @bchocho